repos/fdbt-site/src/pages/api/deletePassenger.ts

```diff
@@ -25,6 +28,30 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
             throw new Error('This individual cannot be deleted because it is in use in a group.');
         }
 
+        const products = await getAllProductsByNoc(nationalOperatorCode)
+        const matchingJsonLinks = products.map((product) => product.matchingJsonLink);
+        const tickets = await Promise.all(
+            matchingJsonLinks?.map(async (link) => {
+                return await getProductsMatchingJson(link);
+            }),
+        );
+
+        const productsUsingPassengerType = tickets?.filter(t => t?.passengerType?.id === id)
+        const productNames = productsUsingPassengerType?.map(ticket=> ticket.products[0] ? (ticket.products[0].productName || 'missing'): "")
+        if(productsUsingPassengerType && productsUsingPassengerType.length > 0){
+            const passengerDetails : any = await getPassengerTypeById(id, nationalOperatorCode)
+            const { name } = passengerDetails || ""
+            const errorMessage = `You cannot delete ${name} because ${productNames.some(p => p!=='missing') ? `it is part of the following product(s): ${productNames?.join(', ')}.`: 'it is in use in a product(s).'} `
+            const errors: ErrorInfo[] = [{ id: '/viewPassengerTypes', errorMessage}];
+            updateSessionAttribute(req, PRODUCTS_USING_PASSENGER_TYPE, {
```

Comment:
instead add the session attribute for the errors ... 
VIEW_PASSENGER_TYPE

and assign the errors in that and use the same in while displaying it

Reply:
[c54320c](https://github.com/fares-data-build-tool/create-data/pull/578/commits/c54320c83560a526531b3cd112f6764e27f23dfe)